### ISSUE TITLE
Not all dependencies are installed or bundled

### ISSUE THREAD

**Issue by KenRmk** [jspm/jspm-cli#1630]
TL;DR:
Some dependencies of modules are not installed or bundled

I encountered some strange behavior when installing and bundling dependencies. First I installed `npm:http-node` (I also installed it with `chrome-net` since it needs a working `net` module but more on this over at #1629) which requires `buffer` in one of its files but that did not get installed.
Using bundle `jspm bundle http-node + chrome-net bundle-all.js` only resulted in error.
```
system-csp-production.js:4 XHR finished loading: GET "chrome-extension://ajogfihphkfpepplepgkgcblkgpmmidd/jspm_packages/npm/buffer@4.5.0.json".N @ system-csp-production.js:4(anonymous function) @ system-csp-production.js:4(anonymous function) @ system-csp-production.js:4(anonymous function) @ system-csp-production.js:4(anonymous function) @ system-csp-production.js:5(anonymous function) @ system-csp-production.js:5(anonymous function) @ system-csp-production.js:5(anonymous function) @ system-csp-production.js:5(anonymous function) @ system-csp-production.js:5(anonymous function) @ system-csp-production.js:4
index.js:11 Uncaught ReferenceError: require is not defined(anonymous function) @ index.js:11
system-csp-production.js:4 Uncaught (in promise) Error: chrome-extension://ajogfihphkfpepplepgkgcblkgpmmidd/jspm_packages/npm/buffer@4.5.0/index.js did not call System.register or AMD define. If loading a global module configure the global name via the meta exports property for script injection support.
	Error loading chrome-extension://ajogfihphkfpepplepgkgcblkgpmmidd/jspm_packages/npm/buffer@4.5.0/index.js as "buffer" from chrome-extension://ajogfihphkfpepplepgkgcblkgpmmidd/dist/main.js
    at t (chrome-extension://ajogfihphkfpepplepgkgcblkgpmmidd/jspm_packages/system-csp-production.js:4:61)
    at g (chrome-extension://ajogfihphkfpepplepgkgcblkgpmmidd/jspm_packages/system-csp-production.js:4:9910)
    at chrome-extension://ajogfihphkfpepplepgkgcblkgpmmidd/jspm_packages/system-csp-production.js:4:8050
```
Looking into the `bundle-all.js` revealed presence of only `nodelibs-buffer` and not the browserify one.
```
System.registerDynamic("github:jspm/nodelibs-buffer@0.1.0.json", ...
System.registerDynamic("github:jspm/nodelibs-buffer@0.1.0/index.js", ...
```
Bundling with `buffer` didnt't really help `jspm bundle http-node + chrome-net + buffer bundle-all.js`

So i tried bundling `buffer` separately `jspm bundle buffer bundle-buffer.js`. That results in nearly empty `bundle-buffer.js` containing only `github:jspm/nodelibs-buffer@0.2.0-alpha.json`.
Then i tried installing `jspm install buffer` and bundling `buffer` again and this is where it got strange because the first time i tried it worked out OK and `bundle-buffer.js` was full of all the code an everything whereas the second time (trying these steps to be sure I'm posting relevant issue) it ended up with almost empty file again.
However after sucessfully installing the buffer I tried to bundle everything again and that event crashed the bundler
```
jspm bundle http-node + chrome-net bundle-all.js
     Building the bundle tree for http-node + chrome-net...

err  Error on fetch for github:jspm/nodelibs-buffer@0.1.0/global.js at /jspm_packages/github/jspm/nodelibs-buffer@0.1.0/global.js
  Loading npm:http-node@1.0.0/_http_outgoing.js
  Loading npm:http-node@1.0.0/http.js
  Error: ENOENT: no such file or directory, open '\jspm_packages\github\jspm\nodelibs-buffer@0.1.0\global.js'
    at Error (native)
```

I'm sorry for the verbosity. I'm trying to give a detailed insight into problems that's been causing me some major headaches over the past two weeks.

**Comment by guybedford** [comment]
I'd strongly advise using jspm 0.17 if you're looking for better npm library support. Handling more edge cases here was a large part of that work.

**Comment by guybedford** [comment]
Oh I see it looks like you are actually using jspm 0.17...

Note that for jspm 0.17 the browserify core libs like `jspm/nodelibs-buffer` should be installed to edge and not from their stable vesrions (which are the jspm 0.16 versions). Try `jspm install buffer --edge`.

**Comment by guybedford** [comment]
Any progress here?

**Comment by KenRmk** [comment]
I'm sorry I've been very busy for the past few weeks. My problem was that not all of the dependencies of the module were installed. In my example simply installing `chrome-net` didn't install the `buffer` with it. At least in the time of creating this issue.

**Comment by guybedford** [comment]
Thanks for the update, I just checked it again here and it definitely is installing Buffer with the current beta. Let me know if you have further issues though.
